libcontainer/cgroups/fs/hugetlb_test.go

```diff
@@ -32,38 +31,47 @@ func TestHugetlbSetHugetlb(t *testing.T) {
 		hugetlbAfter  = 512
 	)
 
-	helper.writeFileContents(map[string]string{
-		limit: strconv.Itoa(hugetlbBefore),
-	})
-
-	helper.CgroupData.c.HugetlbLimit = []*configs.HugepageLimit{
-		{
-			Pagesize: hugePageSize[0],
-			Limit:    hugetlbAfter,
-		},
-	}
-	hugetlb := &HugetlbGroup{}
-	if err := hugetlb.Set(helper.CgroupPath, helper.CgroupData.c); err != nil {
-		t.Fatal(err)
+	for _, pageSize := range HugePageSizes {
+		helper.writeFileContents(map[string]string{
+			fmt.Sprintf(limit, pageSize): strconv.Itoa(hugetlbBefore),
+		})
 	}
 
-	value, err := getCgroupParamUint(helper.CgroupPath, limit)
-	if err != nil {
-		t.Fatalf("Failed to parse %s - %s", limit, err)
+	for _, pageSize := range HugePageSizes {
+		helper.CgroupData.c.HugetlbLimit = []*configs.HugepageLimit{
+			{
+				Pagesize: pageSize,
+				Limit:    hugetlbAfter,
+			},
+		}
+		hugetlb := &HugetlbGroup{}
+		if err := hugetlb.Set(helper.CgroupPath, helper.CgroupData.c); err != nil {
+			t.Fatal(err)
+		}
 	}
-	if value != hugetlbAfter {
-		t.Fatalf("Set hugetlb.limit_in_bytes failed. Expected: %v, Got: %v", hugetlbAfter, value)
+
+	for _, pageSize := range HugePageSizes {
+		limit := fmt.Sprintf(limit, pageSize)
+		value, err := getCgroupParamUint(helper.CgroupPath, limit)
+		if err != nil {
+			t.Fatalf("Failed to parse %s - %s", limit, err)
+		}
+		if value != hugetlbAfter {
+			t.Fatalf("Set hugetlb.limit_in_bytes failed. Expected: %v, Got: %v", hugetlbAfter, value)
+		}
 	}
 }
 
 func TestHugetlbStats(t *testing.T) {
 	helper := NewCgroupTestUtil("hugetlb", t)
 	defer helper.cleanup()
-	helper.writeFileContents(map[string]string{
-		usage:    hugetlbUsageContents,
-		maxUsage: hugetlbMaxUsageContents,
-		failcnt:  hugetlbFailcnt,
-	})
+	for _, pageSize := range HugePageSizes {
+		helper.writeFileContents(map[string]string{
+			fmt.Sprintf(usage, pageSize):    hugetlbUsageContents,
+			fmt.Sprintf(maxUsage, pageSize): hugetlbMaxUsageContents,
+			fmt.Sprintf(failcnt, pageSize):  hugetlbFailcnt,
+		})
+	}
```

Comment:
I don't think test all HugetlbSizes for all test cases are necessary. The problem is `GetStats` will get all HugetlbSizes but we didn't create them in the fake cgroup directory. Can we just create all HugetlbSizes in this test case?